velox/functions/lib/string/StringImpl.h

```diff
@@ -319,6 +319,18 @@ hmacSha512(TOutString& output, const TInString& key, const TInString& data) {
       folly::ByteRange((const uint8_t*)data.data(), data.size()));
 }
 
+// Compute the HMAC-MD5 Hash.
+template <typename TOutString, typename TInString>
+FOLLY_ALWAYS_INLINE void
+hmacMd5(TOutString& output, const TInString& key, const TInString& data) {
+  output.resize(16);
+  folly::ssl::OpenSSLHash::hmac(
+      folly::MutableByteRange((uint8_t*)output.data(), output.size()),
+      EVP_md5(),
+      folly::ByteRange((const uint8_t*)key.data(), key.size()),
```

Comment:
This code repeats a lot in this file. Consider extracting a helper function.

```
folly::ssl::OpenSSLHash::hmac(
   mutableRange(output),
   EVP_md5(),
   range(key),
   range(data));
```

Reply:
This function hmac() is used only once directly in this file. The folly::ssl::OpenSSLHash module has helpers for all other functions. Please suggest where can I add a helper function for this?

Reply:
@nmahadevuni I suggest to add two helper functions:

```
template <typename TString>
inline folly::ByteRange range(const TString& value) {
  return folly::ByteRange((const uint8_t*)value.data(), value.size());
}

template <typename TString>
inline folly::MutableByteRange mutableRange(TString& value) {
  return folly::MutableByteRange((uint8_t*)value.data(), value.size());
}
```

Then, use them in multiple places in this file:

```
/// Compute the SHA256 Hash.
template <typename TOutString, typename TInString>
FOLLY_ALWAYS_INLINE void sha256(TOutString& output, const TInString& input) {
  output.resize(32);
  folly::ssl::OpenSSLHash::sha256(mutableRange(output), range(input));
}

/// Compute the SHA512 Hash.
template <typename TOutString, typename TInString>
FOLLY_ALWAYS_INLINE void sha512(TOutString& output, const TInString& input) {
  output.resize(64);
  folly::ssl::OpenSSLHash::sha512(mutableRange(output), range(input));
}

...
```

Reply:
Sorry, totally misread your first comment. Made the changes.